qiskit/pulse/utils.py

```diff
@@ -16,9 +16,62 @@
 Pulse utilities.
 """
 import warnings
+
+from collections import namedtuple
+from typing import List
+
+from .exceptions import PulseError
+
 # pylint: disable=unused-argument
 
 
+Interval = namedtuple('Interval', ['start', 'stop'])
+"""Intervals [start, stop) are exclusive of the stop time."""
+
+
+def overlaps(first: Interval, second: Interval) -> bool:
+    """
+    Return True iff first and second overlap.
+
+    Note: first.stop may equal second.start, since Interval stop times are exclusive.
+    """
+    if first.start == second.start == second.stop:
+        # They fail to overlap if one of the intervals has duration 0
+        return False
+    if first.start > second.start:
+        first, second = second, first
+    return second.start < first.stop
+
+
+def insertion_index(intervals: List[Interval], new_interval: Interval, index: int = 0) -> int:
```

Comment:
Not explicitly -- any time instructions are added to a schedule, this is implicitly tested. I was lax about it before because there are many existing tests like this. I agree that these should be tested specifically; I'll add the tests